File: tests/fixtures/python/project2/projects.py
from core import AuditLog, Entity, Repository
from users import Team, User


class Task(Entity):
    def __init__(self, entity_id: str, assignee: User):
        super().__init__(entity_id)
        self.assignee: User = assignee
        self.audit: AuditLog = AuditLog()

    def key(self) -> str:
        return self.entity_id

    def reassign(self, user: User) -> None:
        self.assignee = user


class Project(Entity):
    def __init__(self, entity_id: str, owner: User, team: Team):
        super().__init__(entity_id)
        self.owner: User = owner
        self.team: Team = team
        self.tasks: list[Task] = []

    def key(self) -> str:
        return self.entity_id

    def add_task(self, task: Task) -> None:
        self.tasks.append(task)

    def archive(self, repository: Repository) -> None:
        repository.save(self)

File: tests/fixtures/java/project2/Projects.java
import java.util.ArrayList;
import java.util.List;

class Task extends Entity {
    private User assignee;
    private AuditLog audit = new AuditLog();

    public Task(String id, User assignee) {
        super(id);
        this.assignee = assignee;
    }

    public String key() { return id; }
    public void reassign(User user) { this.assignee = user; }
}

class Project extends Entity {
    private User owner;
    private Team team;
    private List<Task> tasks = new ArrayList<Task>();

    public Project(String id, User owner, Team team) {
        super(id);
        this.owner = owner;
        this.team = team;
    }

    public String key() { return id; }
    public void addTask(Task task) { tasks.add(task); }
    public void archive(Repository repository) { repository.save(this); }
}
